pkg/operator/operator.go

```diff
@@ -139,6 +151,10 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		// Do not return the error, it would degrade the whole cluster.
 		// Remember the error and put it in Available condition message below.
 		c.lastError = err
+
+		// TODO: If cluster is blocked from upgrades we should run these checks more often
+		blockUpgrade, blockUpgradeReason = c.checkForDeprecation(util.VSphereClusterInfo)
```

Comment:
I was thinking a check could return some special `UnupgradeableError` that would tell `sync()` to mark the cluster un-upgradeable. We would need a new cluster check with vCenter version. And perhaps rearrange our alerts, because  `CollectNodeHWVersion` check would start failing, raising an alert about it, while we already have another alert about the HW version.

Reply:
I have pushed something simpler for now: 

```golang
blockUpgrade, blockUpgradeReason = c.checkForDeprecation(util.VSphereClusterInfo)
		// if we blocked upgrades then we should retry sooner
		if blockUpgrade {
			delay = c.backoff.Step()
		}
```